core/src/main/scala/cats/instances/hash.scala

```diff
@@ -0,0 +1,23 @@
+package cats
+package instances
+
+import cats.kernel.instances._
+
+trait HashInstances {
+  implicit val catsContravariantCartesianForHash: ContravariantCartesian[Hash] =
+    new ContravariantCartesian[Hash] {
+      /**
+       * Derive an `Hash` for `B` given an `Hash[A]` and a function `B => A`.
+       */
+      def contramap[A, B](ha: Hash[A])(f: B => A): Hash[B] = Hash.by(f)(ha)
```

Comment:
this is untested

Reply:
Thanks. will be tested

Reply:
never mind, i've found it in `cats.tests`.